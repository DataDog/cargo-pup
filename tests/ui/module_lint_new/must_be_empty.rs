//@compile-flags: --crate-name test_must_be_empty
//@compile-flags: --crate-type lib

// Empty module (allowed)
pub mod empty_module {
    // This module is intentionally empty and should not trigger errors
}

// Another empty module (also allowed)
pub mod another_empty_module {
    // This module is also intentionally empty
}

// Non-empty module (should trigger an error)
pub mod non_empty_module { //~ ERROR: Module must be empty
    pub fn invalid_function() { //~ ERROR: Item 'invalid_function' not allowed in empty module
        println!("This module should be empty");
    }
    
    pub const INVALID_CONST: &str = "Content not allowed"; //~ ERROR: Item 'INVALID_CONST' not allowed in empty module
} 