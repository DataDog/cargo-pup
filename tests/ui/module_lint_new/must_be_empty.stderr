error: Module must be empty
  --> tests/ui/module_lint_new/must_be_empty.rs:15:1
   |
LL | / pub mod non_empty_module {
LL | |     pub fn invalid_function() {
LL | |         println!("This module should be empty");
...  |
LL | |     pub const INVALID_CONST: &str = "Content not allowed";
LL | | } 
   | |_^
   |
   = help: Remove all content from this module
   = note: Applied by cargo-pup rule 'must_be_empty_test'.
   = note: `#[deny(module_must_be_empty_lint)]` on by default

error: aborting due to 1 previous error

